pkg/controller/elasticsearch/nodespec/podspec.go

```diff
@@ -112,7 +112,8 @@ func BuildPodTemplateSpec(
 		WithVolumes(volumes...).
 		WithVolumeMounts(volumeMounts...).
 		WithInitContainers(initContainers...).
-		WithInitContainerDefaults(corev1.EnvVar{Name: settings.HeadlessServiceName, Value: headlessServiceName}).
+		// inherit all env vars from main containers to allow Elasticsearch tools that read ES config to work in initContainers
+		WithInitContainerDefaults(builder.MainContainer().Env...).
```

Comment:
nit: not sure if it's really important in that context but `builder.MainContainer()` may return `nil`.

Reply:
That's true but given that we construct the pod builder ourselves a few lines above and ensure through defaulting that we have a main container I would say we can maybe ignore it here?